sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -333,6 +333,16 @@ def _parse_odata_body(error_format, response):
             pass
         return None
 
+    def __str__(self):
+        retval = super(HttpResponseError, self).__str__()
+        try:
+            body = self.response.text()
+            if body and not self.error:
```

Comment:
If `self.error` evaluates to true, does that mean it's already included in `retval`?

Reply:
Yep, if `self.error` exists, it means the error body was correctly parsed with odatav4 (code [here](https://github.com/Azure/azure-sdk-for-python/blob/5c006b5b53a6bfade999b8f0a90ab4c7a9fff8e1/sdk/core/azure-core/azure/core/exceptions.py#L310)). In that case, the relevant error body details are already parsed out and included in the str error response